### ISSUE TITLE
panic: no supported languages found []string{"zh_TW.UTF-8"}

### ISSUE THREAD

**Issue by markkwsu** [cloudfoundry/cli#846]
High five for submitting a report!
To avoid additional back and forth, please include the following details as relevant:

### Command

cf

### CLI Version

6.18.0_osx

### CC API Endpoint Version

see `cf api` output

### CF Trace

For bugs related to HTTP requests, please run the command with env var `CF_TRACE=true`

### Platform Details

Mac OS X 10.11

### Shell

Terminal

**Comment by challiwill** [comment]
@markkwsu Can you please include the commands you ran to get the panic? What did you set the locale to (`cf config --locale xx-XX`)?

**Comment by markkwsu** [comment]
@challiwill I didn't include any commands. So I can't do `cf config --locale xx-XX` too. I will check my environment, maybe it needs to be clean up.

**Comment by challiwill** [comment]
Oh, so you just ran `cf` and you got that panic? Which download link did you use?

**Comment by markkwsu** [comment]
Yes, I use this one: [release](https://cli.run.pivotal.io/stable?release=macosx64&version=6.18.1&source=github-rel).
I also tried the [binary](https://cli.run.pivotal.io/stable?release=macosx64-binary&version=6.18.1&source=github-rel) and the error is the same.

**Comment by challiwill** [comment]
@markkwsu Okay I think you must have an incorrect config file.

If you look at your `~.cf/config.json` I would expect you will see a field `"Locale": "zh-TW"` which is not a supported locale. 

You can either set delete the file (it will get regenerated when you next run a command), or set the locale to one of the following:
```
de-DE
en-US
es-ES
fr-FR
it-IT
ja-JP
ko-KR
pt-BR
zh-Hans
zh-Hant
```

Let me know if that works.

**Comment by markkwsu** [comment]
@challiwill My config.json Locale setting is empty, like:
"Locale": ""

**Comment by markkwsu** [comment]
@challiwill I delete config.json and execute cf command again:
`panic: no supported languages found []string{"zh_TW.UTF-8"}`
I think you guys should consider about fix this issue...

**Comment by challiwill** [comment]
Do you have a "CF_HOME" environment variable set to something other than the usual home dir?

**Comment by markkwsu** [comment]
No.
![image](https://cloud.githubusercontent.com/assets/7066739/15547187/4f436a82-22d5-11e6-8c03-a4948f644e54.png)

**Comment by challiwill** [comment]
Oh interesting. I think the cli actually tries to set its language based on the `LANG` env var, it's specified in these [really hidden docs](https://github.com/cloudfoundry/cli/blob/master/cf/i18n/README-i18n.md). We should probably move that to somewhere more obvious. 

If you set the locale explicitly in your config.json does it work? 

If that's the case we will consider how to change this so the user experience is better.

**Comment by markkwsu** [comment]
@challiwill After I set Locale, and it works.

![image](https://cloud.githubusercontent.com/assets/7066739/15560002/cb87fb5a-2319-11e6-8425-8657d3ab4e41.png)

And `cf` execute result:

![image](https://cloud.githubusercontent.com/assets/7066739/15560013/f160a6c4-2319-11e6-9f36-27124897ec1a.png)

I hope you'll consider to fix this, because "zh-Hant" or "zh-Hans" are strange locale setting for me.
My experience is, these should be "zh-TW" or "zh-HK" for Traditional Chinese, and "zh-CN" for Simply Chinese.
And this issue does not exist under Windows environment, only MacOS.

**Comment by challiwill** [comment]
@markkwsu Thanks for trying it, and I'm glad it worked. 

We'll take a look at changing the locale. This is the first we've heard that the current configuration is irregular. @dkoper do you want to keep this issue open while considering the proposed UX change, or have it closed if @markkwsu thinks we've adequately addressed it.

**Comment by challiwill** [comment]
@dkoper on further consideration it looks like maybe this should be investigated as a separate bug because it looks like our code is supposed to handle "zh-TW" and "zh-HK" but apparently doesn't right now in some cases ( https://github.com/cloudfoundry/cli/blob/master/cf/i18n/i18n.go#L50-L52 ).

Also perhaps instead of panic-ing we should have it default to a language and suggest the user sets the locale with `cf config --locale xx-XX`?

@markkwsu which OS version are running it on?

**Comment by markkwsu** [comment]
@challiwill 
![image](https://cloud.githubusercontent.com/assets/7066739/15563112/32a62ad2-2338-11e6-9099-92db032f42d4.png)

**Comment by dkoper** [comment]
Sounds like we have a few issues:
1. cf CLI responds to LANG (at least on some platforms) and that's not documented
2. When LANG is set to a locale code we don't support, we display what looks like a debugging message
3. We document "zh_Hans" and "zh_Hant" as the supported locale codes, while @markkwsu notes that's weird and we should use "zh-TW" and "zh-HK".
4. We have code to handle "zh-TW" and "zh-HK" that seems not to work.

I'd like to split this out into individual stories - I'll update this issue report as I create them.

**Comment by dkoper** [comment]
We have fixed point 4 above in cf CLI 6.19.0, released today.
https://github.com/cloudfoundry/cli#downloads

Regards,
Dies Koper
CF CLI PM
